### ISSUE TITLE
About ExecutionResult

### ISSUE THREAD

**Issue by gregorvilkner** [graphql-dotnet/graphql-dotnet#1844]
First off, congrats on the release. It's a big one - I use this library in plugins of desktop applications and versioning conflicts with Newtonsoft have caused lots of head scratching in the past. Using System.Text.Json has immediately shown good results.

## Summary

Two things: 

Per graphql spec, shouldn't the "data" and "errors" attributes serialize in lower case? Using graphiql - my introspection queries don't come in as schema docs using "Data".

I've had issues with really long introspection load times, or "no schema available" messages. I'm positive my schema isn't out of the ordinary, at about 200 lines. Digging deeper I found that at times the "document" attribute of the ExecutionResult throws errors when it's serialized. I'm leaving it out for now and things are peachy. What is the reason for having this attribute?

### Environment (if relevant)

I use graphql schema first in controllers of .net framework and .net core / blazor web api projects.

**Comment by Shane32** [comment]
Are you using the `GraphQL.SystemTextJson.DocumentWriter` class to serialize the result to json?

See upgrade notes here:
https://graphql-dotnet.github.io/docs/guides/migration3#json-parsing-and-serialization

As long as you are using the provided `IDocumentWriter` implementation, the `ExecutionResult` is serialized via a custom converter within the `ExecutionResultJsonConverter` class, which serializes the `data` and `errors` properties in the correct case, and ignores the `Document` property.

Note that these classes are provided within a separate nuget package available here:
https://www.nuget.org/packages/GraphQL.SystemTextJson/

Additional serialization configuration options are available for errors - see docs here:
https://graphql-dotnet.github.io/docs/getting-started/errors

**Comment by gregorvilkner** [comment]
Oh. Excellent. You got pixidust in that converter... I noticed the usage in your web api example. Will have to try that. This should resolve a lot of issues I've been having.

I looked it up, I think I had been going with something like this: https://dev.to/dotnet/how-you-can-build-a-web-api-using-graphql-net-core-and-entity-framework-1ago and there the result goes straight into the return of the api call. Which works - but more out of dumb luck it seams.

Thanks for clarifying this. Cheers, Gregor
